### ISSUE TITLE
Regression: "bad extension" when connecting to httpbin.org:443

### ISSUE THREAD

**Issue by yan12125** [openssl/openssl#2133]
Log of s_client on 6dd083fd6804a3ee6ac3adc019f81910f1c63f21 (git bisect points to this commit):
```
$ /home2/yen/sysroot/usr/local/bin/openssl s_client -connect httpbin.org:443
CONNECTED(00000005)
140243072000064:error:141B306E:SSL routines:tls_collect_extensions:bad extension:ssl/statem/extensions.c:318:
140243072000064:error:141710E3:SSL routines:tls_process_server_hello:parse tlsext:ssl/statem/statem_clnt.c:1312:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 66 bytes and written 183 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID: 
    Session-ID-ctx: 
    Master-Key: 
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1482412929
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
---
```

**Comment by davidben** [comment]
The problem seems to be that `TLSEXT_TYPE_ec_point_formats` is missing a `EXT_TLS1_2_SERVER_HELLO` flag.

**Comment by yan12125** [comment]
> The problem seems to be that ```TLSEXT_TYPE_ec_point_formats``` is missing a ```EXT_TLS1_2_SERVER_HELLO``` flag.

Thanks this patch works for me on master:
```Diff
diff --git a/ssl/statem/extensions.c b/ssl/statem/extensions.c
index 760b150a6..f6b0589ff 100644
--- a/ssl/statem/extensions.c
+++ b/ssl/statem/extensions.c
@@ -120,7 +120,7 @@ static const EXTENSION_DEFINITION ext_defs[] = {
 #ifndef OPENSSL_NO_EC
     {
         TLSEXT_TYPE_ec_point_formats,
-        EXT_CLIENT_HELLO | EXT_TLS1_2_AND_BELOW_ONLY,
+        EXT_CLIENT_HELLO | EXT_TLS1_2_SERVER_HELLO | EXT_TLS1_2_AND_BELOW_ONLY,
         NULL, tls_parse_ctos_ec_pt_formats, tls_parse_stoc_ec_pt_formats,
         tls_construct_stoc_ec_pt_formats, tls_construct_ctos_ec_pt_formats,
         final_ec_pt_formats
```
Is it correct? I'm not familiar with TLS protocols.

**Comment by mattcaswell** [comment]
Fix in #2153

**Comment by yan12125** [comment]
Works with the current master, thanks you all!
